contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java

```diff
@@ -284,10 +337,58 @@ public DoubleValueObserver getDoubleValueObserver(
    * @param name - metric name
    * @param description metric description
    * @param unit - metric unit
+   * @param updater - the value updater
    * @return new or memoized {@link LongValueObserver}
    */
   public LongValueObserver getLongValueObserver(
-      final String name, final String description, final String unit) {
-    return meter.longValueObserverBuilder(name).setDescription(description).setUnit(unit).build();
+      final String name,
+      final String description,
+      final String unit,
+      final Consumer<AsynchronousInstrument.LongResult> updater) {
+    return meter
+        .longValueObserverBuilder(name)
+        .setDescription(description)
+        .setUnit(unit)
+        .setUpdater(
+            proxiedLongObserver(name, description, unit, InstrumentType.VALUE_OBSERVER, updater))
+        .build();
+  }
+
+  private Consumer<AsynchronousInstrument.DoubleResult> proxiedDoubleObserver(
+      String name,
```

Comment:
Do you want these to be `final` for consistency? (same below)

Reply:
Definitely, updated.